x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/enrollment/tool/BootstrapPasswordAndEnrollmentTokenForInitialNode.java

```diff
@@ -0,0 +1,199 @@
+import org.elasticsearch.xpack.security.tool.HttpResponse;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.security.SecureRandom;
+import java.util.function.Function;
+
+import static org.elasticsearch.xpack.security.tool.CommandLineHttpClient.createURL;
+
+public class BootstrapPasswordAndEnrollmentTokenForInitialNode extends KeyStoreAwareCommand {
+    private static final String elasticUser = ElasticUser.NAME;
+    private KeyStoreWrapper keyStoreWrapper;
+    private final CheckedFunction<Environment, EnrollmentTokenGenerator, Exception> createEnrollmentTokenFunction;
+    private final Function<Environment, CommandLineHttpClient> clientFunction;
+    private final CheckedFunction<Environment, KeyStoreWrapper, Exception> keyStoreFunction;
+    final SecureRandom secureRandom = new SecureRandom();
+
+    // Package-private for testing
+    SecureString bootstrapPassword;
+    SecureString credentialsPassword;
+
+    CommandLineHttpClient getClient(Environment env) {
+        return clientFunction.apply(env);
+    }
+    EnrollmentTokenGenerator getEnrollmentTokenGenerator(Environment env) throws Exception{
+        return createEnrollmentTokenFunction.apply(env);
+    }
+    KeyStoreWrapper getKeyStoreWrapper(Environment env) throws Exception { return keyStoreFunction.apply(env); }
+    OptionParser getParser() { return parser; }
+
+    BootstrapPasswordAndEnrollmentTokenForInitialNode() {
+        this(
+            environment -> new CommandLineHttpClient(environment),
+            environment -> KeyStoreWrapper.load(environment.configFile()),
+            environment -> new EnrollmentTokenGenerator(environment)
+        );
+    }
+
+    BootstrapPasswordAndEnrollmentTokenForInitialNode(Function<Environment, CommandLineHttpClient> clientFunction,
+                                                      CheckedFunction<Environment, KeyStoreWrapper, Exception> keyStoreFunction,
+                                                      CheckedFunction<Environment, EnrollmentTokenGenerator, Exception>
+                                                     createEnrollmentTokenFunction){
+        super("Set elastic password and generate enrollment token for initial node");
+        this.clientFunction = clientFunction;
+        this.keyStoreFunction = keyStoreFunction;
+        this.createEnrollmentTokenFunction = createEnrollmentTokenFunction;
+        parser.allowsUnrecognizedOptions();
+    }
+
+    public static void main(String[] args) throws Exception {
+        exit(new BootstrapPasswordAndEnrollmentTokenForInitialNode().main(args, Terminal.DEFAULT));
+    }
+
+    @Override
+    protected void execute(Terminal terminal, OptionSet options, Environment env) throws Exception {
+        if (options.nonOptionArguments().contains("--explicitly-acknowledge-execution") == false) {
```

Comment:
https://github.com/elastic/elasticsearch/pull/75816#discussion_r685256471

> But it can be invoked. I don't have strong opinion on this though, but doesn't heart leaving it here

How can it be invoked when we don's ship a CLI tool ? You mean a user writing a script on their own that tries to use this as a main class ? If they go to this extent, what stops them from passing --explicitly-acknowledge-execution as a parameter while they're at it ?